ckanext/ozwillo_organization_api/plugin.py

```diff
@@ -216,3 +221,118 @@ def get_actions(self):
             'create-ozwillo-organization': create_organization,
             'delete-ozwillo-organization': delete_organization
         }
+
+
+# Used for tests purposes
+class CreateOrganizationPlugin(plugins.SingletonPlugin):
+    plugins.implements(plugins.interfaces.IOrganizationController, inherit=True)
+
+    def create(self, entity):
+        after_create(entity, 'Cassis')
+
+
+def after_create(entity, city_name):
+    '''
+    This method is called after a new instance is created.
+    It uses the services from data.gouv.fr to automatically add data to our new instance.
+    It is possible to add automatically any other resources from different services as
+    long as an api returns the desired resources urls.
+
+    :param entity: object, the organization being created
+    :param city_name: string, the name of the city being created. It has to be as close as possible
+    to the real city name as it would be used as the main parameter in the request to retrieve the data
+    :return:
+    '''
+
+    city = slugify(city_name)
+    organization_id = entity.id
+    insee_re = re.compile(r'\d{5}')
+    site_url = config.get('ckan.site_url')
+    base_url_1 = 'https://www.data.gouv.fr/api/1/territory/suggest/?q='
+    base_url_2 = 'https://www.data.gouv.fr/api/1/spatial/zone/{}/datasets?'
+    check_url = site_url + '/api/3/action/package_search?q='
+
+    try:
+        # Get the city from the gouv api and extract the name, id, description and insee
+        city_response = requests.get(base_url_1 + city)
+        city_json = city_response.json()
+        city_name = slugify(city_json[0]['title'])
+        city_id = city_json[0]['id']
+        city_description = city_json[0]['page']
+        city_insee = insee_re.search(city_id).group()
+        log.info(city_name)
+
+        # Check if a package with city_name already exists. If it does, add the date and time to the city name
+        package_exist = requests.get(check_url + city_name)
+        if package_exist.json()['result']['count'] != 0:
```

Comment:
not sure of this thing. it is in the case we have more than one entry for a given organization ?

Reply:
Yes, if I don't rename the organization, the organization creation fail.

Reply:
Hmm, this should not happen. IIRC, CKAN (or this plugin) will refuse to create an organization that already exists ?

Reply:
Yes that is what I meant, that is why I renamed it. I can remove it so when we try to create an organization that already exists, it won't be possible.

Reply:
I prefer raising an error back to Ozwillo if asked to create an organization that already exists (because a given organization is not meant to be declared twice)